openedx/edx-ora2
# Disable uploading files while an upload is in progress

This PR disables the upload button and shows a spinner during uploads so continuing to click the upload button doesn't flood the LMS with requests.

**What changed?**

- Disables the Upload Files button when upload starts.
- Shows a spinner to indicate to the learner an upload is in progress.
- Toggle both when the upload finishes or fails.

**Developer Checklist**


**Testing Instructions**

1. Deploy this version of the xblock to a devstack or instance.
2. Create a course and add the xblock.
3. Edit the xblock settings to allow uploads.
4. Go to the xblock in the LMS.
5. Open the browser developer console and set throttling to 3G.
6. Select a few files and click Upload files.
7. Confirm the button is disabled and a spinner appears.
8. Disable throttling in the developer console.
9. Confirm the button is enabled again and the spinner disappears when the upload finishes.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
